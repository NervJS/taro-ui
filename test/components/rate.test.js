import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtRate from '../../.temp/components/rate/index'

describe('AtRate Snap', () => {
  it('render initial AtRate', () => {
    const componet = renderToString(<AtRate isTest />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRate -- props customStyle', () => {
    const componet = renderToString(<AtRate customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRate -- props className', () => {
    const componet = renderToString(<AtRate className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRate -- props size', () => {
    const componet = renderToString(<AtRate size='10' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRate -- props value', () => {
    const componet = renderToString(<AtRate value='2' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRate -- props max', () => {
    const componet = renderToString(<AtRate max='10' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtRate -- props margin', () => {
    const componet = renderToString(<AtRate margin='10' />)
    expect(componet).toMatchSnapshot()
  })
})
