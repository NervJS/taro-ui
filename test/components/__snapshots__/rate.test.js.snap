// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtRate Snap render AtRate -- props className 1`] = `"<div class=\\"at-rate test\\"><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div></div>"`;

exports[`AtRate Snap render AtRate -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-rate\\"><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div></div>"`;

exports[`AtRate Snap render AtRate -- props margin 1`] = `"<div class=\\"at-rate\\"><div style=\\"margin-right:0.2134rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.2134rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.2134rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.2134rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.2134rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div></div>"`;

exports[`AtRate Snap render AtRate -- props max 1`] = `"<div class=\\"at-rate\\"><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div></div>"`;

exports[`AtRate Snap render AtRate -- props size 1`] = `"<div class=\\"at-rate\\"><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:10px;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div></div>"`;

exports[`AtRate Snap render AtRate -- props value 1`] = `"<div class=\\"at-rate\\"><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--on\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--on\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div></div>"`;

exports[`AtRate Snap render initial AtRate 1`] = `"<div class=\\"at-rate\\"><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div><div style=\\"margin-right:0.1067rem;\\" class=\\"at-rate__icon at-rate__icon--off\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span><div class=\\"at-rate__left\\"><span style=\\"font-size:;\\" class=\\"taro-text at-icon at-icon-star-2\\"></span></div></div></div>"`;
